grafana/client.sls

```diff
@@ -0,0 +1,72 @@
+{%- from "grafana/map.jinja" import client with context %}
+{%- if client.enabled %}
+
+/etc/salt/minion.d/_grafana.conf:
+  file.managed:
+  - source: salt://grafana/files/_grafana.conf
+  - template: jinja
+  - user: root
+  - group: root
+
+{%- for datasource_name, datasource in client.datasource.iteritems() %}
+
+grafana_client_datasource_{{ datasource_name }}:
+  grafana_datasource.present:
+  - name: {{ datasource_name }}
+  - type: {{ datasource.type }}
+  - url: http://{{ datasource.host }}:{{ datasource.get('port', 80) }}
```

Comment:
I'll merge it and propose better solution with new salt resources later 2day with your comments integrated.